src/index/IndexImpl.h

```diff
@@ -0,0 +1,731 @@
+// Copyright 2015, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+#pragma once
```

Comment:
This is the old `Index.h`, if you want to you can `vimdiff` the two files somewhere locally,
Unfortunately this is not shown here.
Again I have marked the places where small actual changes happened.